extensions/apps/notifications/src/components/utils/notifications-util.tsx

```diff
@@ -0,0 +1,140 @@
+import AppIcon from '@akashaorg/design-system-core/lib/components/AppIcon';
+import {
+  Antenna,
+  Profile,
+  Vibes,
+} from '@akashaorg/design-system-core/lib/components/Icon/akasha-icons';
+import {
+  ChannelOptionIndexes,
+  FollowNotificationMetaData,
+  MentionNotificationMetaData,
+  PushOrgNotification,
+  ReflectionNotificationMetaData,
+} from '@akashaorg/typings/lib/sdk';
+import { InboxNotification } from '@akashaorg/typings/lib/ui';
+import { BoltIcon, GlobeAltIcon, RectangleGroupIcon } from '@heroicons/react/24/outline';
+import React from 'react';
+
+import dayjs from 'dayjs';
+import 'dayjs/locale/es';
+import 'dayjs/locale/ro';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import calendar from 'dayjs/plugin/calendar';
+dayjs.extend(relativeTime);
+dayjs.extend(calendar);
+/**
+ * Icons based on the APPs
+ */
+const placeholderIcons = {
+  [ChannelOptionIndexes.ANTENNA]: <Antenna />,
+  [ChannelOptionIndexes.PROFILE]: <Profile />,
+  [ChannelOptionIndexes.VIBES]: <Vibes />,
+};
+
+/**
+ * Transform notification, of type PushOrgNotification, to Inbox Notification
+ * It adds the icons and other information based on the payload of the notification
+ */
+export const getPresentationDataFromNotification = (
+  notification: PushOrgNotification,
+): InboxNotification => {
+  // set Default data
+  const returnObj: InboxNotification = {
+    title: notification.payload.data.asub,
+    body: notification.payload.data.amsg,
+    // Title and icon for broadcast TBD decided in future iterations
+    notificationTypeIcon: <GlobeAltIcon />,
+    notificationTypeTitle: 'BROADCAST',
+    notificationAppIcon: null,
+    ctaLinkTitle: null,
+    ctaLinkUrl: null,
+    date: '',
+    isSeen: !notification.isUnread,
+    appName: '',
+  };
+
+  // Set notification type title and icon
+  switch (notification.payload.data.type) {
+    case 3:
+      returnObj.notificationTypeTitle = 'ACTIVITY';
+      returnObj.notificationTypeIcon = <BoltIcon />;
+      break;
+    case 4:
+      // Title and icon for group TBD decided in future iterations
+      returnObj.ctaLinkTitle = 'GROUP';
+      returnObj.notificationTypeIcon = <RectangleGroupIcon />;
+  }
+
+  const parsedMetaData = notification.payload.data.parsedMetaData;
+  // set CTA/button title and url to navigate
+  switch (returnObj.title) {
+    case 'New mention':
```

Comment:
can we find a better field to use for branching? from what i see `notification.payload.data` already contains an `app` field. maybe we can use that?

Reply:
![Screenshot 2024-12-06 at 11 29 35](https://github.com/user-attachments/assets/6bd97626-d5ee-49f5-8455-fdb48f973c1b)
The app represents the Apps the user has subscribed to. This means the user can subscribe to other Apps other than 'Akasha'. The notification title is set in the [API](https://github.com/AKASHAorg/api/blob/d87281af00084d20ef98eddfa35e2e1cb0f68c12/apps/indexing/src/queue/create-job.ts) have a look at it by searching the 'NotificationPayload'